hazelcast/src/main/java/com/hazelcast/internal/server/tcp/TcpServerConnectionManagerBase.java

```diff
@@ -113,14 +119,88 @@ void refreshNetworkStats() {
     }
 
     static class Plane {
-        final ConcurrentHashMap<Address, TcpServerConnection> connectionMap = new ConcurrentHashMap<>(100);
-        final Set<Address> connectionsInProgress = newSetFromMap(new ConcurrentHashMap<>());
         final ConcurrentHashMap<Address, TcpServerConnectionErrorHandler> errorHandlers = new ConcurrentHashMap<>(100);
         final int index;
 
+        private final Set<Address> connectionsInProgress = newSetFromMap(new ConcurrentHashMap<>());
+        private final ConcurrentHashMap<Address, TcpServerConnection> connectionMap = new ConcurrentHashMap<>(100);
+
         Plane(int index) {
             this.index = index;
         }
+
+        TcpServerConnection getConnection(Address address) {
+            return connectionMap.get(address);
+        }
+
+        void putConnection(Address address, TcpServerConnection connection) {
+            connectionMap.put(address, connection);
+        }
+
+        void putConnectionIfAbsent(Address address, TcpServerConnection connection) {
+            connectionMap.putIfAbsent(address, connection);
+        }
+
+        void removeConnection(TcpServerConnection connection) {
+            removeConnectionInProgress(connection.getRemoteAddress());
+            connectionMap.entrySet().removeIf(entry -> entry.getValue().equals(connection));
```

Comment:
`removeIf` implementation is affected by https://bugs.java.com/bugdatabase/view_bug.do?bug_id=8078645 which is not fixed in any JDK 8 release as far as I can tell. Best to follow an approach similar to `removeConnectionsWithId`.